app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/commontask/CommonTaskViewModel.kt

```diff
@@ -413,4 +462,19 @@ class CommonTaskViewModel(appComponent: AppComponent = TaigaApp.appComponent) :
             session.taskEdit.postUpdate()
         }
     }
+
+    // Copy link
+    fun setCopyLinkInClipboard() {
```

Comment:
As I said above, all interactions with the system services should be done from the view part, not from the view model. Generally, I would suggest the following: just add new field (call it `url`, for example) in the `CommonTaskExtended` class. And then add this calculation [here](https://github.com/EugeneTheDev/TaigaMobile/blob/master/app/src/main/kotlin/io/eugenethedev/taigamobile/data/repositories/TasksRepository.kt#L319). Thus the code would be much more concise